app/javascript/packs/foremMobile.js

```diff
@@ -0,0 +1,114 @@
+/* global Runtime, fetchBaseData */
+
+function initializeNamespaceWhenPageIsReady() {
+  setTimeout(() => {
+    // Wait for data-loaded so we can ensure initializers have executed. This
+    // way we know the Runtime class will be available globally
+    if (document.body.getAttribute('data-loaded') === 'true') {
+      // We're ready to initialize
+      if (Runtime.currentMedium() === 'ForemWebView') {
+        loadForemMobileNamespace();
+      }
+    } else {
+      // Page hasn't initialized yet. We need to wait until the page is ready
+      initializeNamespaceWhenPageIsReady();
+    }
+  }, 100);
+}
+
+function loadForemMobileNamespace() {
+  window.ForemMobile = {
+    retryDelayMs: 700,
+    getUserData() {
+      const userStatus = document
+        .getElementsByTagName('body')[0]
+        .getAttribute('data-user-status');
+      if (userStatus === 'logged-in') {
+        return document
+          .getElementsByTagName('body')[0]
+          .getAttribute('data-user');
+      }
+    },
+    getInstanceMetadata() {
+      return JSON.stringify({
+        name: document.querySelector("meta[property='forem:name']")['content'],
+        logo: document.querySelector("meta[property='forem:logo']")['content'],
+        domain: document.querySelector("meta[property='forem:domain']")[
+          'content'
+        ],
+      });
+    },
+    registerDeviceToken(token, appBundle, platform) {
+      const params = JSON.stringify({
+        token,
+        platform,
+        app_bundle: appBundle,
+      });
+      const csrfToken = document.querySelector(
+        "meta[name='csrf-token']",
+      )?.content;
+      fetch('/users/devices', {
```

Comment:
Refactor (blocking): Use the [request](https://github.com/forem/forem/blob/main/app/javascript/utilities/http/request.js) utility function. It handles the CSRF token and has sensible defaults which are the ones used in the fetch call here.

Reply:
Looks like exactly what I needed in the first place, will do. Thanks for the suggestion!